pkg/sql/distsqlrun/indexbackfiller.go

```diff
@@ -185,6 +185,9 @@ func (ib *indexBackfiller) runChunk(
 				ib.rowVals, secondaryIndexEntries); err != nil {
```

Comment:
The comment above EncodeSecondaryIndexes doesnt say that it can generate a different number of index entries as the indexes passed in. This contract has been completely broken. I'd ask that you change the comment to explain that while it's expected to be the same size passed in, it will return a different size slice. I also feel we should panic at the start of EncodeSecondaryIndexes() for len(indexes) == len(secondaryIndexes)

and a comment above the ```if len(entries) > 1 {```

and while you're at it can you change 

```
if len(entries) > 1 {
 +			secondaryIndexEntries = append(secondaryIndexEntries, entries[1:]...)
 +		}
 +		if len(entries) > 0 {
 +			secondaryIndexEntries[i] = entries[0]
```
to

```
secondaryIndexEntries[i] = entries[0]
if len(entries) > 1 {
 			secondaryIndexEntries = append(secondaryIndexEntries, entries[1:]...)
 		}
```

Reply:
Do you think adding unittests to catch the same problem for all the different call sites of this function are valuable?